fpdc_client/base.py

```diff
@@ -188,9 +189,13 @@ def create(cls, data, server=None):
 
         server = server or _SERVER
         _check_server(server)
-        result = server.client.action(
-            server.schema, [cls.api_endpoint, "create"], params=data
-        )
+        try:
+            result = server.client.action(
+                server.schema, [cls.api_endpoint, "create"], params=data
+            )
+        except coreapi.exceptions.LinkLookupError:
+            print(f"You are not allowed to create a {cls.api_endpoint}")
+            sys.exit(1)
```

Comment:
Since it's a library we can't assume the environment where it will be used. It may be a command line script but it may also be inside a webserver, or an async job processor. I think we shouldn't call `sys.exit()` here but instead raise our own exception with this nice message. I would create a base `FPDCError(Exception)` class and a `PermissionDenied(FPDCError)` class to make it easier for users to know what happened.

Reply:
Ha yes, I faced this case while writing a script so the sys.exit made sense in that situation but will not in all cases.